src/appsettings/validators.py

```diff
@@ -67,3 +69,53 @@ def __call__(self, value):
             if not isinstance(element, self.value_type):
                 params = {"key": key, "value": element, "type": self.value_type.__name__}
                 raise ValidationError(self.message, params=params)
+
+
+class FileValidator(object):
+    """Validator which checks file existence and read, write permissions."""
+
+    DEFAULT_ERROR_MESSAGES = {
+        "exists": "File %(value)s does not exist.",
+        "read_perm": "No read permission for %(value)s.",
+        "write_perm": "No write permission for %(value)s.",
+    }
+
+    def __init__(self, exists=True, read_perm=True, write_perm=False, error_messages=None):
+        """
+        Validator initialization.
+
+        Args:
+            exists (bool): Check whether or not does the file exist.
+            read_perm (bool): Check whether or not do we have read permissions.
+            write_perm (bool): Check whether or not do we have write permissions.
+            error_messages (dict): Update default error messages. Each message must contain the ``%(value)s``. Possible
+                keys are ``exists``, ``read_perm`` and ``write_perm`` (corresponds to previous arguments and their
+                checks).
+        """
+        self.exists = exists
+        self.read_perm = read_perm
+        self.write_perm = write_perm
+        self.error_messages = {**self.DEFAULT_ERROR_MESSAGES, **(error_messages or {})}
+
+    @classmethod
+    def _get_nearest_existing_directory(cls, filename):
+        """Get nearest existing directory of file ``filename``."""
+        directory = os.path.dirname(os.path.abspath(filename))
+        if not os.path.exists(directory):
+            return cls._get_nearest_existing_directory(directory)
+        return directory
+
+    def __call__(self, value):
+        """Validate the ``value``."""
+        if not os.path.exists(value):
+            if self.exists:
+                raise ValidationError(self.error_messages["exists"], params={"value": value})
+            # If file does not exist (yet), we might want to check at least permissions to the directory, where the file
+            # will be located.
+            value = self._get_nearest_existing_directory(value)
```

Comment:
This seems to be rather far-fetched. We have no idea what are the requirements for the directories up in hierarchy. I'd rather skip the check entirely and leave it on a custom validator if required.

Reply:
Also, once we start looking at the directories up, we need to check completely different set of permissions.

Reply:
I don't understand why is it so bad and which different permissions should we check? When you want to read/write a new file in a directory, you just need read/write permission in the directory, or not?

So what do you suggest? Skip the `read_perm` and `write_perm` when the file does not exist without any message to the user?

Reply:
If I need to read a file that may not exist, I need to check the write permission on its directory if I want to create it.

Reply:
But you don't want to create it, when you want to just read it. When you want to have read permission on an non-existing file, you just expect that a file with read permission will have been created when you try to read it. Read permission on an existing parent directory does not ensure it, but it is a good sign.

Reply:
Fixed according to our consultation.